src/mctpd.c

```diff
@@ -1790,6 +1790,31 @@ static int query_get_peer_uuid(peer *peer) {
 	return rc;
 }
 
+static int interface_call_to_ifindex_busowner(ctx *ctx, sd_bus_message *msg)
+{
+	const char *iface, *path;
+	link_userdata *link;
+	int rc, ifindex;
+
+	path = sd_bus_message_get_path(msg);
+	if (!path)
+		return -1;
+
+	rc = sd_bus_path_decode_many(path, MCTP_DBUS_PATH_LINKS "/%", &iface);
+	if (rc <= 0)
+		return -1;
+
+	ifindex = mctp_nl_ifindex_byname(ctx->nl, iface);
+	if (ifindex < 0)
+		return -1;
+
+	link = mctp_nl_get_link_userdata(ctx->nl, ifindex);
```

Comment:
How about using `mctp_nl_get_link_userdata_byname`. So we can directly use `iface`.

Reply:
We need the ifindex as well though.

Reply:
I agree.